### PR TITLE
[SPARK-42667][CONNECT] Spark Connect: newSession API

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

This PR proposes an implementation of newSession API. The idea is we reuse user context(e.g. user_id), gRPC channel, etc. But differentiate different Spark Remote Session by client id, which is generated randomly. 

So this idea has the benefits of:
1. reusing gRPC channel to not over too manny connections to the server.
2. Each user can has multiple remote sessions, differentiated by client ids (or named session ids in server side).  

### Why are the changes needed?

API coverage

### Does this PR introduce _any_ user-facing change?

NO
### How was this patch tested?

UT

### CODE REVIEW

**Comment by amaliujia** [comment]
@hvanhovell

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala [hunk 1]
```diff
@@ -344,7 +344,9 @@ class SparkSession private[sql] (
   // scalastyle:on
 
   def newSession(): SparkSession = {
-    throw new UnsupportedOperationException("newSession is not supported")
+    val newClient: SparkConnectClient =
```
Comment (hvanhovell) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala:347]: Small nit. We could also implement `SparkConnectClient.copy()`. That way we don't have to expose as much of the client.

**Review by hvanhovell — COMMENTED** [review]
One small nit. Looks good otherwise!

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala (hunk 1 above)
Comment (amaliujia) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala]: This is a very good idea.

**Review by hvanhovell — APPROVED** [review]
LGTM

**Pull request closed without merging**